### PR TITLE
Preventing crash when post.status is nil.

### LINKED ISSUE #3522: Crash occurs when post status is missing/nil.
As seen in feature request #2420, posts with custom statuses do not work correctly on mobile. When using a Jetpack enabled site and the JSON API will return post object JSON without a `status` string. Post objects within the app will have a `nil` status and crash when attempting to edit the status.

### PR DESCRIPTION
**Pull request by kurzee**
Resolves #3522

### CODE REVIEW

**Comment by bummytime** [comment]
@kurzee Your logic will will fix the crasher...thanks!  It is a decent stopgap for the time being. Can you update this branch with the latest from develop and then I will merge it in if everything works.

File: WordPress/Classes/ViewRelated/Post/PostSettingsViewController.m [hunk 1]
```diff
@@ -757,12 +757,17 @@ - (void)showPostStatusSelector
...
+    }
+       
     NSDictionary *statusDict = @{
                                  @"DefaultValue": NSLocalizedString(@"Published", @""),
                                  @"Title" : NSLocalizedString(@"Status", nil),
                                  @"Titles" : titles,
-                                 @"Values" : statuses,
-                                 @"CurrentValue" : self.apost.status
+                                 @"Values" : titles,
+                                 @"CurrentValue" : postStatusTitle
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/Post/PostSettingsViewController.m]: This change will result in the title of a status (e.g. Published, Pending Review, Draft) being assigned where a status const value (e.g. publish, pending, draft) is expected.

**Comment by aerych** [comment]
Something looked a bit off so I dug into the history of this code a bit.  

It looks like we'd started using the status title as the thing to edit back in 67c36fd8a6a386bdf83af149d459d81d92762c50 (or maybe earlier I stopped looking at this point).  Things were shuffled a bit in a477db200916fd52dd974c043861442d93eba98a when I did some clean up on the post settings vc. Then in f1eb8274758a570c2d7ca2278fee3533ceaa936c I switched the code away from transforming title strings to post status, to just use the status consts as cannonical f1eb8274758a570c2d7ca2278fee3533ceaa936c.  

In f1eb8274758a570c2d7ca2278fee3533ceaa936c I missed updating the `DefaultValue` to be the `PostStatusPublished` const, (mea culpa!) so that's a problem I need to correct. 

We really shouldn't rely on title strings (especially localized ones) to match post status values, so I think we need a different approach. 

If the problem is that the post status coming back from a REST request is `nil`, then it might make more sense to patch PostService and ensure `status` is never `nil`. If we get an unsupported custom status, the convention is to assign it to `PostStatusDraft`.

**Comment by kurzee** [comment]
@aerych Great! In this case the PR can be closed and I will create a new PR handling nil statuses when the object itself is created.

**Comment by aerych** [comment]
> In this case the PR can be closed and I will create a new PR handling nil statuses when the object itself is created

Sounds good.  Thanks @kurzee!

**Comment by kurzee** [comment]
@aerych this is no longer an issue with the latest changes in develop. No additional PR or tweaks needed.

**Pull request closed without merging**